src/ClientFactory.php

```diff
@@ -34,14 +38,33 @@ class ClientFactory {
    *   The Drupal-configured HTTP handler stack.
    * @param \Drupal\Core\Http\ClientFactory $httpClientFactory
    *   The Drupal core HTTP client factory.
+   * @param \Drupal\Core\Cache\CacheBackendInterface $cacheBackend
+   *   The cache used for mpx responses.
    */
-  public function __construct(HandlerStack $handlerStack, HttpClientFactory $httpClientFactory) {
+  public function __construct(HandlerStack $handlerStack, HttpClientFactory $httpClientFactory, CacheBackendInterface $cacheBackend) {
     $this->httpClientFactory = $httpClientFactory;
     $this->handlerStack = $handlerStack;
 
     // This must be in the constructor, otherwise this will be added every time
     // a new mpx client is created during a page request.
     $this->handlerStack->push(Middleware::mpxErrors(), 'mpx_errors');
+
+    $cache = new DrupalGuzzleCache($cacheBackend);
+
+    $strategy = new DelegatingCacheStrategy();
+    $strategy->registerRequestMatcher(new MpxObjectRequestMatcher(), new MpxCacheStrategy($cache, 3600 * 24 * 30));
```

Comment:
By default, we cache objects for 30 days. Sites should get fresh data by listening to notifications.